### PR TITLE
Add fwgettext util function

### PR DESCRIPTION
**Pull request by rubenwardy**
Replaces manual use of `wgettext` and `wsprintf` with a new util function `fwgettext`. This improves code quality

## To do

This PR is Ready for Review.

- [ ] Replace all uses of `gettext` with versions that return strings, so they can be RAII deleted?

### CODE REVIEW

**Review by SmallJoker — APPROVED** [review]
There's two more in `guiVolumeChange.cpp` if you'd like to change them.
The hardcoded limit of 255 characters is not that great, but good enough for now.

LGTM

File: src/gettext.h [hunk 1]
```diff
@@ -59,3 +59,21 @@ inline std::string strgettext(const std::string &text)
...
+ * @return translated string
+ */
+template <typename ...Args>
+inline std::wstring fwgettext(const char *src, Args&&... args)
+{
+	wchar_t buf[255];
+	const wchar_t* str = wgettext(src);
+	swprintf(buf, sizeof(buf) / sizeof(wchar_t), str, std::forward<Args>(args)...);
+	delete[] str;
+	return std::wstring(buf);
```
Comment (sfan5) [inline src/gettext.h:78]: You could fix the hardcoded limit like this:
```c++
    std::wstring ret;
    const wchar_t* str = wgettext(src);
    const int required_size = swprintf(nullptr, 0, str, std::forward<Args>(args)...);
    if (required_size > 0) {
        ret.resize(required_size);
        swprintf(&ret[0], ret.size(), str, std::forward<Args>(args)...);
    }
    delete[] str;
    return ret;
```

Comment (Desour) [reply src/gettext.h:78]: According to https://en.cppreference.com/w/cpp/io/c/fwprintf:
> ### Notes
>While narrow strings provide std::snprintf, which makes it possible to determine the required output buffer size, there is no equivalent for wide strings, and in order to determine the buffer size, the program may need to call std::swprintf, check the result value, and reallocate a larger buffer, trying again until successful.

Also, I think you'd have an off-by-one error, as you need a size of 1 greater that `reqired_size` (for the 0).

(Oh, and you could wrap `str` into a `std::unique_ptr<wchar_t[]>`.)

Comment (sfan5) [reply src/gettext.h:78]: Hmm, I thought it would behave identically. That's unfortunate..

**Comment by sfan5** [comment]
Also an `fgettext` that does the same for non-wide chars would be useful (plus it should use `mt_snprintf`). I count at least 4 cases where this would be useful.

**Review by nerzhul — APPROVED** [review]

**Pull request merged**